soroban-env-host/src/host.rs

```diff
@@ -2448,6 +2442,25 @@ impl VmCallerEnv for Host {
         Ok(RawVal::VOID)
     }
 
+    fn update_current_contract_wasm(
```

Comment:
Should this host function emit a system event indicating that an update happened? This would allow downstream systems to listen for implementation changes.

Reply:
Added a system event. I'm a bit uncertain about its contents, but we don't have any convention for the system events yet, so it's hard to say what's right.

Reply:
I don't see the implementation of `emit_update_contract_event` anywhere. And re: convention, I think an empty data with `topics = ["wasm_update", old_executable, new_executable]` should be fine.

Reply:
Sorry, forgot to add the file. It should be pushed now.

As for the topics, your proposal won't quite work as IIUC topics need to be flat and contain a limited number of elements, which doesn't allow for using enums. I don't quite understand the significance of topics TBH, but I think here it shouldn't matter much that we put the update context to data.

Reply:
We no longer have the limitation for topics to be flat, but it's fine for this info to be in data. As for the significance of topics, it's just a way to organize info in an event similar to the EVM where you have some data that you can filter by. We could instead have events be a single ScVal, but downstream systems like soroban-rpc would have to come up with a way to filter within an ScVal.

Reply:
Have we actually merged the change for removing the topic restrictions? If not, I'd just merge this - it's not hard to update this if needed.

Reply:
Ok, I was under the impression this is not supported (probably docs are outdated). Updated.